keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,375 @@
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I need the ability to add/remove Network Interfaces to
+existing VMs without re-creating the VM. This would not be applicable for the
+Primary Interface of the Pod.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary Interface of the Pod.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will produce a separate
+KEP with detailed design for specified set of requirements. Each of this KEP
+will have to take all of the requirements, from each phase, into consideration
+when created.
+
+#### Phase I (base API and reference in Pod)
+1. This effort shall not change the behavior of Today existing clusters
+2. We need to introduce an “Object” that represents the existing 
+infrastructure’s networking
+3. “Object” is decoupled from the network definition/description created by the
+network administrator
+4. The “Object” shall not define any implementation specific parameters in that
+object
+5. “Object” shall provide option to define:
+   * IPAM mode: external/internal
+   * List of route prefixes - optional and not forced on the implementations
+6. “Object” can reference to implementation-specific parameters
+7. “Object” is the consumer-facing declaration for workloads
+8. Cluster “Default” “Object” is the network the cluster has been created with
+9. Cluster “Default” “Object” cannot be changed or removed
+10. Pods shall reference the “object” when trying to attach to the specific
+networks
+    * Pods has to specify all the “object” that it wishes to attach to,
+      including the Cluster “Default” “Object”
+11. Workloads can access “Object” when “attach” RBAC exists at the attachment time
+    * “attach” RBAC is new verb to allow referencing API objects
+12. The Pod reference to a Networks is optional and when NOT specified, Pod
+connects to “Default” “Object” (network the cluster has been created with)
+13. Pod shall be able to provide additional configuration on how it attaches to
+a network
+    * Identify what “Object” is the Primary network
+    * Optional parameters: MAC address, IP address, speed, MTU, interface name etc.
+14. Every Pod connected to specific network (represented by the “Object”) must
```

Comment:
Is this a necessary limitation for all "Objects"? Perhaps a network is meant only for north-south traffic. I wonder if this could be left to be defined by the implementation, similarly to the next point.

Reply:
I think that would be too open. Today's K8s network has exactly this requirement, this way we standardize across all the networks.

Do you have a use case for such situation?